test/integration/mapping_validation_test.rb

```diff
@@ -20,6 +20,35 @@ class MappingValidationTest < ActiveSupport::TestCase
       #{invalid_categories}"
   end
 
+  test "every Shopify category has corresponding channel mappings" do
+    mappings_json_data = CLI.new.parse_json("dist/en/integrations/all_mappings.json")
+    shopify_category_lack_mappings = []
+
+    mappings_json_data["mappings"].each do |mapping|
+      next unless mapping["input_taxonomy"].include?("shopify")
+
+      all_shopify_category_ids = category_ids_from_taxonomy(mapping["input_taxonomy"])
+      next if all_shopify_category_ids.nil?
+
+      shopify_category_ids_from_mappings_input = Set.new
+      mapping["rules"].each do |rule|
+        shopify_category_ids_from_mappings_input.add(rule["input"]["category"]["id"])
+      end
```

Comment:
Can we do this as an operation instead of an each?
```suggestion
      shopify_category_ids_from_mappings_input = mapping["rules"]
        .map { _1.dig("input", "category", "id") }
        .to_set
```